sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -366,11 +370,15 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       return df
     }
 
-    // replacementMap is either Map[String, String] or Map[Double, Double] or Map[Boolean,Boolean]
-    val replacementMap: Map[_, _] = replacement.head._2 match {
-      case v: String => replacement
-      case v: Boolean => replacement
-      case _ => replacement.map { case (k, v) => (convertToDouble(k), convertToDouble(v)) }
+    // replacementMap is either Map[String, String], Map[Double, Double], Map[Boolean,Boolean]
+    // while value can have null
```

Comment:
If the types are not these three types, what are the behaviors? Could you explain them here? Also, please add negative examples too. Thanks~